test/box/broken_snapshot.test.lua

```diff
@@ -0,0 +1,123 @@
+
+-- write data recover from latest snapshot
+env = require('test_run')
+test_run = env.new()
+
+test_run:cmd("restart server default")
+fio = require 'fio'
+test_run:cmd("setopt delimiter ';'")
+function get_snap_file ()
+    local snapfile = nil	
+    local directory = fio.pathjoin(fio.cwd(), 'broken_snapshot')
+    for files in io.popen(string.format("ls %s", directory)):lines() do
+        local snaps = string.find(files, "snap")
+        if (snaps ~= nil) then
+            local snap = string.find(files, "%n")
+	    if (snap ~= nil) then
+                snapfile = string.format("%s/%s", directory, files)
+            end
+        end
+    end
+    return snapfile
+end;
+test_run:cmd("setopt delimiter ''");
+
+
+test_run:cmd("create server test with script='box/broken_snapshot.lua'")
+test_run:cmd("start server test")
+test_run:cmd("switch test")
+space = box.schema.space.create('test', { engine = "memtx" })
+space:format({ {name = 'id', type = 'unsigned'}, {name = 'year', type = 'unsigned'} })
+index = space:create_index('primary', { parts = {'id'} })
+
+for key = 1, 10000 do space:insert({key, key + 1000}) end
+box.snapshot()
+
+test_run:cmd("switch default")
+snapfile = get_snap_file()
+file = io.open(snapfile, "r")
+size = file:seek("end")
+if size > 30000 then size = 30000 end
+
+-- save last snapshot
+os.execute(string.format('cp %s %s.save', snapfile, snapfile))
+-- write garbage at the end of file
+file = io.open(snapfile, "ab")
```

Comment:
What about more sophisticated snap file contamination? Like change space id to non-existent, insert garbage in the middle of snap, corrupt system space etc

Reply:
about system space Mons write in some comment before that we not need to boot when system space was corrupted, about change space id, I try to add this case in test